src/include/kompute/TypeContainer.hpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "ABCTypeContainer.hpp"
+#include <typeinfo>
+
+
+struct IdCounter
+{
+    static size_t counter;
+};
+
+template<typename T>
+class TypeContainer : public ABCTypeContainer, IdCounter
+{
+  private:
+    size_t classId()
+    {
+        static size_t id = counter++;
+        return id;
+    }
+
+  public:
+    TypeContainer() : dt(typeid(T)) {}
```

Comment:
Why is a classid counter necessary on top of this?

Reply:
see above